src/aws_encryption_sdk/keyring/raw_keyring.py

```diff
@@ -0,0 +1,221 @@
+        if not plaintext_data_key:
+            return EncryptKeyError("Unable to generate data encryption key.")
+
+        # plaintext_data_key to RawDataKey
+        data_encryption_key = RawDataKey(key_provider=key_provider, data_key=plaintext_data_key)
+
+        # Add generated data key to encryption_materials
+        encryption_materials.add_data_encryption_key(data_encryption_key, encryption_materials.keyring_trace)
+
+    else:
+        plaintext_data_key = encryption_materials.data_encryption_key
+
+    # Encrypt data key
+    encrypted_wrapped_key = wrapping_key.encrypt(
+        plaintext_data_key=plaintext_data_key, encryption_context=encryption_materials.encryption_context
+    )
+
+    # EncryptedData to EncryptedDataKey
+    encrypted_data_key = aws_encryption_sdk.internal.formatting.serialize.serialize_wrapped_key(
+        key_provider=key_provider,
+        wrapping_algorithm=wrapping_algorithm,
+        wrapping_key_id=key_name,
+        encrypted_wrapped_key=encrypted_wrapped_key,
+    )
+
+    # Add encrypted data key to encryption_materials
+    encryption_materials.add_encrypted_data_key(encrypted_data_key, encryption_materials.keyring_trace)
+
+    return encryption_materials
+
+
+def on_decrypt_helper(decryption_materials, wrapping_key, wrapping_algorithm, key_name, encrypted_data_key):
+    # Check if plaintext data key exists
+    if decryption_materials.data_key:
+        return decryption_materials
+
+    # Wrapped EncryptedDataKey to deserialized EncryptedData
+    encrypted_wrapped_key = aws_encryption_sdk.internal.formatting.deserialize.deserialize_wrapped_key(
+        wrapping_algorithm=wrapping_algorithm, wrapping_key_id=key_name, wrapped_encrypted_key=encrypted_data_key
+    )
+
+    # EncryptedData to raw key string
+    plaintext_data_key = wrapping_key.decrypt(
+        encrypted_wrapped_data_key=encrypted_wrapped_key, encryption_context=decryption_materials.encryption_context
+    )
+
+    # Update decryption materials
+    data_encryption_key = DataKey(
+        key_provider=encrypted_data_key.key_provider,
+        data_key=plaintext_data_key,
+        encrypted_data_key=encrypted_data_key.encrypted_data_key,
+    )
+    decryption_materials.add_data_encryption_key(data_encryption_key, decryption_materials.keyring_trace)
+
+    return decryption_materials
+
+
+@attr.s
+class RawAESKeyring(Keyring):
+    """Public class for Raw AES Keyring.
+
+    :param str key_namespace: String defining the keyring ID
+    :param bytes key_name: Key ID
+    :param wrapping_key: Encryption key with which to wrap plaintext data key
+    :type wrapping_key: WrappingKey
+    :param wrapping_algorithm: Wrapping Algorithm with which to wrap plaintext data key
+    :type wrapping_algorithm: WrappingAlgorithm
+    """
+
+    key_namespace = attr.ib(validator=attr.validators.instance_of(six.string_types))
+    key_name = attr.ib(hash=True, validator=attr.validators.instance_of(six.binary_type))
```

Comment:
Don't mess with the `hash` settings for any `attr.ib` calls. We set those elsewhere for legacy compatibility reasons, but we should not set them anymore now that we require a newer version of `attrs`.